hadoop-ozone/s3gateway/src/test/java/org/apache/hadoop/ozone/s3/TestUgiFilter.java

```diff
@@ -0,0 +1,171 @@
+    omService = new Text("127.0.0.1:9862");
+    LocalDate now = LocalDate.now();
+    curDate = DATE_FORMATTER.format(now);
+
+    headers = new HashMap<>();
+    parameters = new HashMap<>();
+
+    headers.put(HOST_HEADER, "localhost:9878");
+    headers.put(ENCODE_HEADER, "identity");
+    headers.put(USER_AGENT_HEADER, "aws-cli/2.1.29 "+
+        "Python/3.8.8 Linux/4.15.0-144-generic "+
+        "exe/x86_64.ubuntu.18 prompt/off "+
+        "command/s3api.create-bucket");
+    headers.put(DATE_HEADER, "20210616T195044Z");
+    headers.put(ENCRYPT_TYPE_HEADER,
+        "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855");
+    headers.put(AUTHORIZATION_HEADER, "AWS4-HMAC-SHA256 "+
+        "Credential=AKIAIIIE56NH5ZHKLTWQ/20210616/us-east-1/s3/aws4_request, "+
+        "SignedHeaders=host;x-amz-content-sha256;x-amz-date, "+
+        "Signature=c29b4c46e825d5df56cdde12a61adfa65560a54"+
+        "7c8973b9809621086727a2f2e");
+    headers.put(LENGTH_HEADER, "0");
+
+  }
+
+  @Test
+  public void testUgiFilterDoFilter() throws IOException, ServletException {
+    UgiFilter filter = new UgiFilter();
+    filter.setOzoneConfiguration(this.conf);
+    filter.setOmService(this.omService);
+
+    HttpServletRequest request = Mockito.mock(HttpServletRequest.class);
+    HttpServletResponse response = Mockito.mock(HttpServletResponse.class);
+    FilterChain filterChain = Mockito.mock(FilterChain.class);
+    FilterConfig filterConfig = Mockito.mock(FilterConfig.class);
+
+    // correct date in autherization header for AuthorizationV4QueryParser
+    // date validator
+    headers.put(AUTHORIZATION_HEADER, headers.
+        get(AUTHORIZATION_HEADER).replace("20210616", curDate));
+    Enumeration<String> headerNames = Collections.enumeration(headers.keySet());
+
+    Mockito.when(request.getScheme()).thenReturn("http");
+    Mockito.when(request.getMethod()).thenReturn("PUT");
+    Mockito.when(request.getPathInfo()).thenReturn("/bucket1");
+    Mockito.when(request.getHeaderNames()).thenReturn(headerNames);
+    Mockito.when(request.getHeader(HOST_HEADER)).thenReturn(
+        headers.get(HOST_HEADER));
+    Mockito.when(request.getHeader(ENCODE_HEADER)).thenReturn(
+        headers.get(ENCODE_HEADER));
+    Mockito.when(request.getHeader(USER_AGENT_HEADER)).thenReturn(
+        headers.get(USER_AGENT_HEADER));
+    Mockito.when(request.getHeader(DATE_HEADER)).thenReturn(
+        headers.get(DATE_HEADER));
+    Mockito.when(request.getHeader(ENCRYPT_TYPE_HEADER)).thenReturn(
+        headers.get(ENCRYPT_TYPE_HEADER));
+    Mockito.when(request.getHeader(AUTHORIZATION_HEADER)).thenReturn(
+        headers.get(AUTHORIZATION_HEADER));
+    Mockito.when(request.getHeader(LENGTH_HEADER)).thenReturn(
+        headers.get(LENGTH_HEADER));
+    Mockito.when(request.getParameterMap()).thenReturn(parameters);
+
+    filter.init(filterConfig);
+    filter.doFilter(request, response, filterChain);
+    filter.destroy();
+  }
```

Comment:
Can you please help me understand what is being asserted here?

Reply:
@adoroszlai I reviewed with @neils-dev and confirmed that there was an assert missing which I've added.  Thanks for noticing!

Reply:
Thanks.  Would it make sense to add one or more other test cases where the filter is expected to reject the request(s)?  I guess the current test would pass even with a no-op filter.

Reply:
More tests added @adoroszlai  Sorry it took so long.